psi4/driver/driver.py

```diff
@@ -2050,51 +2049,27 @@ def molden(wfn, filename=None, density_a=None, density_b=None, dovirtual=None):
 
     """
 
-    if filename is None:
-        filename = core.get_writer_file_prefix(wfn.molecule().name()) + ".molden"
+    warnings.warn(
+        "Using `psi4.molden` instead of `wfn.write_molden` is deprecated, and as soon as 1.12 it will stop working\n",
+        category=FutureWarning)
 
-    if dovirtual is None:
-        dovirt = bool(core.get_option("SCF", "MOLDEN_WITH_VIRTUAL"))
+    if density_b and not density_a:
+        raise ValidationError(f"psi4.molden can't receive a beta but not an alpha spindensity.")
 
+    if not density_a:
+        wfn.write_molden(filename, dovirtual, False)
     else:
-        dovirt = dovirtual
-
-    if density_a:
-        nmopi = wfn.nmopi()
-        nsopi = wfn.nsopi()
-
-        NO_Ra = core.Matrix("NO Alpha Rotation Matrix", nmopi, nmopi)
-        NO_occa = core.Vector(nmopi)
-        density_a.diagonalize(NO_Ra, NO_occa, core.DiagonalizeOrder.Descending)
-        NO_Ca = core.Matrix("Ca Natural Orbitals", nsopi, nmopi)
-        NO_Ca.gemm(False, False, 1.0, wfn.Ca(), NO_Ra, 0)
-
+        wfn_ = core.Wavefunction.build(wfn.molecule(), core.get_global_option('BASIS'))
+        wfn_.deep_copy(wfn)
+        wfn_.Da().copy(density_a)
+        wfn_.Da().back_transform(wfn_.Ca())
         if density_b:
-            NO_Rb = core.Matrix("NO Beta Rotation Matrix", nmopi, nmopi)
-            NO_occb = core.Vector(nmopi)
-            density_b.diagonalize(NO_Rb, NO_occb, core.DiagonalizeOrder.Descending)
-            NO_Cb = core.Matrix("Cb Natural Orbitals", nsopi, nmopi)
-            NO_Cb.gemm(False, False, 1.0, wfn.Cb(), NO_Rb, 0)
-
+            wfn_.Db().copy(density_b)
+            wfn_.Db().back_transform(wfn_.Cb())
         else:
-            NO_occb = NO_occa
-            NO_Cb = NO_Ca
-
-        mw = core.MoldenWriter(wfn)
-        mw.write(filename, NO_Ca, NO_Cb, NO_occa, NO_occb, NO_occa, NO_occb, dovirt)
+            wfn_.Db().copy(wfn_.Da())
+        wfn_.write_molden(filename, dovirtual, True)
```

Comment:
(comment has no relation to line of code)

there's a couple places in the docs to amend (search "molden"):
* https://github.com/psi4/psi4/blob/master/doc/sphinxman/source/psithoninput.rst
* doc/sphinxman/source/molden.rst (you can try autofunction psi4.core.Wavefunction.write_molden